packages/aws-cdk/lib/api/hotswap/monitor/logs-monitor.ts

```diff
@@ -0,0 +1,192 @@
+import * as util from 'util';
+import * as colors from 'colors/safe';
+import { error } from '../../../logging';
+import { ISDK } from '../../aws-auth';
+
+// how often we should read events from CloudWatchLogs
+const SLEEP = 2_000;
+
+/**
+ * Represents a CloudWatch Log Event that will be
+ * printed to the terminal
+ */
+export interface CloudWatchLogEvent {
+  readonly message: string;
+  readonly logGroup: string;
+  readonly timestamp: Date;
+}
+
+/**
+ * Options for creating a CloudWatchLogEventMonitor
+ */
+export interface CloudWatchLogEventMonitorOptions {
+  /**
+   * The time that watch was first triggered.
+   * CloudWatch logs will be filtered using this start time
+   *
+   * @default - the current time
+   */
+  readonly hotswapTime?: Date;
+
+  /**
+   * The EventPrinter to use to print out CloudWatch events
+   *
+   * @default - a default printer will be created
+   */
+  readonly printer?: IEventPrinter;
+}
+
+export class CloudWatchLogEventMonitor {
+  /**
+   * Determines which events not to display
+   */
+  private startTime: number;
+
+  private logGroups = new Set<string>();
+  private sdk?: ISDK;
+
+  /**
+   * The event printer that controls printing out
+   * CloudWatchLog Events
+   */
+  public printer: IEventPrinter;
+
+  constructor(options: CloudWatchLogEventMonitorOptions = {}) {
+    this.startTime = options.hotswapTime?.getTime() ?? Date.now();
+    this.printer = options.printer ?? new EventPrinter({
+      stream: process.stderr,
+    });
+  }
+
+  public start(): CloudWatchLogEventMonitor {
```

Comment:
Am I correct in thinking that this only returns `this` because we want to write this line in `CdkToolkit`?

```ts
    const logMonitor = options.logs !== false ? new CloudWatchLogEventMonitor({ hotswapTime: new Date() }).start() : undefined;
```

If that's the case, let's just make it return `void` like the other methods, and have an explicit call to `.start()` in its own line. Otherwise, the call is too "buried" in other code.

And actually, do we need `start()` at all? Can't we just start the monitor when we create it?